### ISSUE TITLE
`fromJsonString()` should represent numbers in the JSON as float values

### ISSUE THREAD

**Issue by MaryamZi** [ballerina-platform/ballerina-lang#24069]
**Description:**

The spec says

```ballerina
# Parses a string in JSON format and returns the the value that it represents.
# All numbers in the JSON will be represented as float values.
# Returns an error if the string cannot be parsed.
#
# + str - string in JSON format
# + return - `str` parsed to json or error
public function fromJsonString(string str) returns json|error = external;
```

See https://github.com/ballerina-platform/ballerina-spec/blob/master/lang/lib/value.bal#L173

We don't seem to be doing this atm.

**Steps to reproduce:**

```ballerina
import ballerina/io;

public function main() {
    string s = "{\"a\": 1, \"b\": \"str\"}";

    map<json> j = <map<json>> s.fromJsonString();

    io:println(j["a"] is int);      // TRUE!
    io:println(j["a"] is float);    // FALSE!

    io:println(j["b"] is string);   // TRUE
}
```

**Affected Versions:**
1.2.4

**Comment by chiranSachintha** [comment]
This issue fix with this PR. https://github.com/ballerina-platform/ballerina-lang/pull/25378

**Comment by MaryamZi** [comment]
Closing this issue since it has been fixed with #25378 according to the updated spec.
